### PR TITLE
fix: webexapi fix for pmr request

### PR DESCRIPTION
**Pull request by amritesi**
Jira card link: [https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-274567](url)

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 1]
```diff
@@ -116,8 +118,31 @@ MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.peopl
 
 
 MeetingInfoUtil.getDestinationType = async (from) => {
-  const {destination, type, webex} = from;
-
+  const {type, webex} = from;
+  let {destination} = from;
+
+  if (type === _PERSONAL_ROOM_) {
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:124]: Can you write comments here on what exactly it does

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 2]
```diff
@@ -116,8 +118,31 @@ MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.peopl
...
+    if (!destination) {
+      destination = {
+        userId: webex.internal.device.userId,
+        orgId: webex.internal.device.orgId
+      };
+    }
+    else {
+      const options = VALID_EMAIL_ADDRESS.test(destination) ? {email: destination} : {id: destination};
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:132]: same here about we assuming the userId as default

File: packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting-info/utilv2.js [hunk 3]
```diff
@@ -68,6 +69,76 @@ describe('plugin-meetings', () => {
...
+          orgId = base64.fromBase64url(orgId);
+          orgId = orgId.split('/');
+          orgId = orgId[orgId.length - 1];
+
+          userId = base64.fromBase64url(userId);
+          userId = userId.split('/');
+          userId = userId[userId.length - 1];
+
+
+          expect(userId).toEqual(mockedListReturn.userId);
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting-info/utilv2.js:118]: This needs to be the converted from base64 encoded

File: packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting-info/utilv2.js [hunk 4]
```diff
@@ -68,6 +69,76 @@ describe('plugin-meetings', () => {
...
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting-info/utilv2.js:138]: same here

**Review by lalli-flores — CHANGES_REQUESTED** [review]
@amritesi please review [commit guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#git-commit-guidelines) so your commits align with our standards.

File: packages/node_modules/@webex/plugin-meetings/src/config.js [hunk 5]
```diff
@@ -88,7 +88,7 @@ export default {
     enableExtmap: false,
     experimental: {
       enableMediaNegotiatedEvent: false,
-      enableUnifiedMeetings: false
+      enableUnifiedMeetings: true
```
Comment (lalli-flores) [inline packages/node_modules/@webex/plugin-meetings/src/config.js:91]: Please remove from code so we can merge

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 6]
```diff
@@ -116,8 +118,32 @@ MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.peopl
 
 
 MeetingInfoUtil.getDestinationType = async (from) => {
-  const {destination, type, webex} = from;
+  const {type, webex} = from;
+  let {destination} = from;
+
+  if (type === _PERSONAL_ROOM_) { // this case checks if your type is personal room
```
Comment (lalli-flores) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:124]: This changes look like they're in the wrong place. Shouldn't they be part of `MeetingInfoUtil.getRequestBody` instead?
That's where the org ID and user ID values need to be figured out

**Review by lalli-flores — CHANGES_REQUESTED** [review]
Make sure that you remove the flag enablement, and that you only have one commit, with the right type and scope. Ex:
```
fix(plugin-meetings): add lookup for PMR meeting info v2

File: packages/node_modules/@webex/plugin-meetings/src/config.js (hunk 5 above)
Comment (amritesi) [reply packages/node_modules/@webex/plugin-meetings/src/config.js:91]: yeah this file was reverted in the last commit

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 7]
```diff
@@ -116,8 +118,32 @@ MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.peopl
...
+    if (!destination) { // if we are not getting anything in desination we fetch org and user ids from webex instance
+      destination = {
+        userId: webex.internal.device.userId,
+        orgId: webex.internal.device.orgId
+      };
+    }
+    else {
+      const options = VALID_EMAIL_ADDRESS.test(destination) ? {email: destination} : {id: destination};// we are assuming userId as default
+      const res = await webex.people.list(options);
```
[the remark this answers is not in the document]
Comment (amritesi) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:133]: options is basically we are expecting the id as default , we are validating the email wether we are getting email or not if not then we are passing the id

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 8]
```diff
@@ -116,8 +118,32 @@ MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.peopl
...
+
+      let {orgId, id: userId} = res.items[0];
+
+      orgId = base64.fromBase64url(orgId);// decoding the org and user ids and filtering the ids out of the decoded url
+      orgId = orgId.split('/');
+      orgId = orgId[orgId.length - 1];
 
+      userId = base64.fromBase64url(userId);
```
[the remark this answers is not in the document]
Comment (amritesi) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:141]: fixed it ,  thanks

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 9]
```diff
@@ -116,8 +118,32 @@ MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.peopl
...
+        orgId: webex.internal.device.orgId
+      };
+    }
+    else {
+      const options = VALID_EMAIL_ADDRESS.test(destination) ? {email: destination} : {id: destination};// we are assuming userId as default
+      const res = await webex.people.list(options);
+
+      let {orgId, id: userId} = res.items[0];
+
+      orgId = base64.fromBase64url(orgId);// decoding the org and user ids and filtering the ids out of the decoded url
```
[the remark this answers is not in the document]
Comment (amritesi) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:137]: fixed it ,thanks

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js (hunk 6 above)
Comment (amritesi) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:124]: @arun3528  not sure of this one .

Comment (lalli-flores) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:123]: We'll leave in on `getDestinationType` for now

**Review by lalli-flores — APPROVED** [review]
Thanks @amritesi, looks good!

**Review by arun3528 — APPROVED** [review]

**Pull request merged**